src/cljs/main/broadfcui/page/method_repo/method/exporter.cljs

```diff
@@ -0,0 +1,204 @@
+(ns broadfcui.page.method-repo.method.exporter
+  (:require
+   [dmohs.react :as react]
+   [broadfcui.common :as common]
+   [broadfcui.common.components :as comps]
+   [broadfcui.common.flex-utils :as flex]
+   [broadfcui.common.icons :as icons]
+   [broadfcui.common.input :as input]
+   [broadfcui.common.links :as links]
+   [broadfcui.common.style :as style]
+   [broadfcui.common.table :refer [Table]]
+   [broadfcui.components.buttons :as buttons]
+   [broadfcui.components.modals :as modals]
+   [broadfcui.components.split-pane :refer [SplitPane]]
+   [broadfcui.components.workspace-selector :refer [WorkspaceSelector]]
+   [broadfcui.endpoints :as endpoints]
+   [broadfcui.page.method-repo.method.common :as method-common]
+   [broadfcui.page.workspace.workspace-common :as ws-common]
+   [broadfcui.net :as net]
+   [broadfcui.utils :as utils]
+   ))
+
+
+(react/defc- Preview
+  {:render
+   (fn [{:keys [state]}]
+     (let [{:keys [config config-error]} @state]
+       (cond config-error (style/create-server-error-message config-error)
+             config [:div {:style {:padding "0.5rem 1rem" :background-color "white"
+                                   :maxHeight "-webkit-fill-available"}}
+                     (method-common/render-config-details config)]
+             :else [comps/Spinner {:text "Loading Configuration Details..."}])))
+   :component-did-mount
+   (fn [{:keys [props this]}]
+     (this :-load (:preview-config props)))
+   :component-will-receive-props
+   (fn [{:keys [props next-props this]}]
+     (when (not= (:preview-config props) (:preview-config next-props))
+       (this :-load (:preview-config next-props))))
+   :-load
+   (fn [{:keys [state]} config]
+     (swap! state dissoc :config :config-error)
+     (let [{:keys [namespace name snapshotId]} config]
+       (endpoints/call-ajax-orch
+        {:endpoint (endpoints/get-configuration namespace name snapshotId true)
+         :on-done (net/handle-ajax-response
```

Comment:
why do we need handle-ajax-response instead of just passing the function?

Reply:
This chunk was largely copy/pasted from elsewhere. I'm not really sure what we want to do with the `net` namespace, it either needs to go away or be made more powerful, right now it's not accomplishing much.

Reply:
`net` was created mainly to define a new pattern. It should be built out.